src/cc/api/BPF.cc

```diff
@@ -164,22 +168,37 @@ StatusTuple BPF::detach_all() {
     return StatusTuple(0);
 }
 
-StatusTuple BPF::attach_kprobe(const std::string& kernel_func,
-                               const std::string& probe_func,
-                               uint64_t kernel_func_offset,
-                               bpf_probe_attach_type attach_type,
-                               int maxactive) {
+StatusTuple BPF::attach_kprobe_(const std::string& kernel_func,
```

Comment:
We never name methods this way. Maybe `attach_kprobe_internal` and make it private